src/distribution-integration/alpm-hook.sh

```diff
@@ -2,12 +2,13 @@
 
 set -euo pipefail
 
-# enumerate files that are managed by holo-files
-holo scan --short | grep '^file:' | cut -d: -f2- | sort > /tmp/files-managed-by-holo
-
-# changed files come from stdin (but without leading slash!); find files
-# managed by Holo among these
-sort | sed 's+^+/+' | comm -12 - /tmp/files-managed-by-holo | sed 's/^/file:/' | xargs -r holo apply
-
-# cleanup
-rm /tmp/files-managed-by-holo
+# ALPM calls this script from the root directory, with respect to the
+# `--root` flag.
+export HOLO_ROOT_DIR="."
```

Comment:
This cannot be left in here. Any non-empty `HOLO_ROOT_DIR`will force plugins into testing mode.

I can certainly see why you did this, though. In testing mode, entity IDs look like `file:etc/foo/bar`, while actual entity IDs in production look like `file:/etc/foo/bar`, with the trailing slash in front. That's a divergence that we should rather address in Holo itself, so that it always uses absolute paths in entity IDs, whether in testing mode or not.

Since the rest of the changes looks good, I will cherry-pick all your commits except for the ones regarding alpm-hook into master now. Please rebase after that.